From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dmitrii <bankersenator@gmail.com>
Date: Thu, 25 Nov 2021 09:44:32 +0000
Subject: [PATCH 051/226] fwb: silence statx boot spam

Signed-off-by: Dmitrii <bankersenator@gmail.com>
---
 libs/hwui/jni/Typeface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/hwui/jni/Typeface.cpp b/libs/hwui/jni/Typeface.cpp
index ce484af149..db483ee9a4 100644
--- a/libs/hwui/jni/Typeface.cpp
+++ b/libs/hwui/jni/Typeface.cpp
@@ -153,7 +153,7 @@ static void Typeface_registerGenericFamily(JNIEnv *env, jobject, jstring familyN
 static bool getVerity(const std::string& path) {
     struct statx out = {};
     if (statx(AT_FDCWD, path.c_str(), 0 /* flags */, STATX_ALL, &out) != 0) {
-        ALOGE("statx failed for %s, errno = %d", path.c_str(), errno);
+        //ALOGE("statx failed for %s, errno = %d", path.c_str(), errno);
         return false;
     }
 
