From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Galt <johngaltfirstrun@gmail.com>
Date: Mon, 18 May 2026 13:01:19 -0400
Subject: [PATCH 09/13] Launcher3: AllAppsList: Move calculateDiff to bg thread

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 .../launcher3/allapps/AlphabeticalAppsList.java  | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/src/com/android/launcher3/allapps/AlphabeticalAppsList.java b/src/com/android/launcher3/allapps/AlphabeticalAppsList.java
index e566ef0ab..0a974c522 100644
--- a/src/com/android/launcher3/allapps/AlphabeticalAppsList.java
+++ b/src/com/android/launcher3/allapps/AlphabeticalAppsList.java
@@ -25,6 +25,8 @@ import static com.android.launcher3.allapps.SectionDecorationInfo.ROUND_BOTTOM_R
 import static com.android.launcher3.allapps.SectionDecorationInfo.ROUND_NOTHING;
 import static com.android.launcher3.logging.StatsLogManager.LauncherEvent.LAUNCHER_PRIVATE_SPACE_PREINSTALLED_APPS_COUNT;
 import static com.android.launcher3.logging.StatsLogManager.LauncherEvent.LAUNCHER_PRIVATE_SPACE_USER_INSTALLED_APPS_COUNT;
+import static com.android.launcher3.util.Executors.DATA_HELPER_EXECUTOR;
+import static com.android.launcher3.util.Executors.MAIN_EXECUTOR;
 
 import android.content.Context;
 import android.text.Spannable;
@@ -357,8 +359,18 @@ public class AlphabeticalAppsList implements AllAppsStore.OnUpdateListener {
         }
 
         if (mAdapter != null) {
-            DiffUtil.calculateDiff(new MyDiffCallback(oldItems, mAdapterItems), false)
-                    .dispatchUpdatesTo(mAdapter);
+            final List<AdapterItem> oldItemsSnapshot = oldItems;
+            final List<AdapterItem> newItemsSnapshot = new ArrayList<>(mAdapterItems);
+            DATA_HELPER_EXECUTOR.execute(() -> {
+                DiffUtil.DiffResult diffResult =
+                        DiffUtil.calculateDiff(
+                                new MyDiffCallback(oldItemsSnapshot, newItemsSnapshot), false);
+                MAIN_EXECUTOR.execute(() -> {
+                    if (mAdapter != null) {
+                        diffResult.dispatchUpdatesTo(mAdapter);
+                    }
+                });
+            });
         }
     }
 
