From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: minaripenguin <minaripenguin@users.noreply.github.com>
Date: Tue, 29 Oct 2024 18:05:36 +0800
Subject: [PATCH] build: Add pihook sysprops

Signed-off-by: minaripenguin <minaripenguin@users.noreply.github.com>
---
 scripts/gen_build_prop.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scripts/gen_build_prop.py b/scripts/gen_build_prop.py
index 04ec410..5790e78 100644
--- a/scripts/gen_build_prop.py
+++ b/scripts/gen_build_prop.py
@@ -126,6 +126,9 @@ def parse_args():
   if config["BuildNumber"].startswith("eng."):
     config["BuildNumber"] = config["DateUtc"]
 
+  config["PihooksGmsFp"] = ""
+  config["PihooksGmsModel"] = ""
+
   override_config(config)
   if args.partition == "system":
     config['ProductBrand'] = config['SystemBrand']
@@ -250,6 +253,9 @@ def generate_build_info(args):
 
   print(f"ro.lineage.device={config['LineageDevice']}")
 
+  print(f"persist.sys.pihooks_FINGERPRINT={config['PihooksGmsFp']}")
+  print(f"persist.sys.pihooks_MODEL={config['PihooksGmsModel']}")
+
   # These values are deprecated, use "ro.product.cpu.abilist"
   # instead (see below).
   print(f"# ro.product.cpu.abi and ro.product.cpu.abi2 are obsolete,")
