From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: israfilbd <144210295+israfilbd@users.noreply.github.com>
Date: Thu, 27 Mar 2025 19:00:56 +0600
Subject: [PATCH 20/23] Disable fsgen check

---
 fsgen/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fsgen/Android.bp b/fsgen/Android.bp
index 773383a7..9d9b948f 100644
--- a/fsgen/Android.bp
+++ b/fsgen/Android.bp
@@ -40,4 +40,5 @@ bootstrap_go_package {
 
 soong_filesystem_creator {
     name: "soong_filesystem_creator",
+    enabled: false,
 }
